src/main/kotlin/ink/abb/pogo/scraper/Bot.kt

```diff
@@ -261,4 +267,28 @@ class Bot(val api: PokemonGo, val settings: Settings) {
     fun task(task: Task) {
         task.run(this, ctx, settings)
     }
+
+    fun checkForPlannedStop():Boolean {
+        val timeDiff:Long = ChronoUnit.MINUTES.between(ctx.startTime, LocalDateTime.now())
+        val pokemonCatched:Int = ctx.pokemonStats.first.get()
+        val pokestopsVisited:Int = ctx.pokestops.get()
+        //Log.red("time: ${timeDiff}, pokemon: ${pokemonCatched}, pokestops: ${pokestopsVisited}")
+        if(settings.botTimeoutAfterMinutes <= timeDiff && settings.botTimeoutAfterMinutes != -1){
+            Log.red("Bot timed out as declared in the settings (after ${settings.botTimeoutAfterMinutes} minutes)")
+            return true
+        } else if(settings.botTimeoutAfterCatchingPokemon <= pokemonCatched && settings.botTimeoutAfterCatchingPokemon != -1){
+            Log.red("Bot timed out as declared in the settings (after catching ${settings.botTimeoutAfterCatchingPokemon} pokemon)")
+            return true
+        } else if(settings.botTimeoutAfterVisitingPokestops <= pokestopsVisited && settings.botTimeoutAfterVisitingPokestops != -1){
+            Log.red("Bot timed out as declared in the settings (after visiting ${settings.botTimeoutAfterVisitingPokestops} pokestops)")
+            return true
+        }
+        return false
+    }
+
+    fun terminateApplication(){
+        phaser.forceTermination()
+        ProgramController.stopAllApplications()
```

Comment:
The Problem with stop is that it is not gracefull stop and you have to kill the process if you want to restart the bot. If you still think it is the correct way i will change it back (i used stop at the beginning but thought it is not Working as expected) and push that